src/fetch.ts

```diff
@@ -32,18 +34,21 @@ export function createFetch ({ fetch }: CreateFetchOptions): $Fetch {
       if (opts.params) {
         request = withQuery(request, opts.params)
       }
-      if (opts.body && typeof opts.body === 'object' && ['patch', 'post', 'put'].includes(opts.method?.toLowerCase() || '')) {
-        try {
-          opts.body = JSON.stringify(opts.body)
-          opts.headers = opts.headers || {}
-          if (Array.isArray(opts.headers)) {
-            opts.headers.push(['content-type', 'application/json'])
-          } else if ('set' in opts.headers) {
-            ;(opts.headers as Headers).set('content-type', 'application/json')
+      if (opts.body && opts.body.toString() === '[object Object]' && payloadMethods.includes(opts.method?.toLowerCase() || '')) {
+        opts.body = JSON.stringify(opts.body)
+        opts.headers = opts.headers || {}
+        if (Array.isArray(opts.headers)) {
+          const index = opts.headers.findIndex(([header]) => header.toLowerCase() === 'content-type')
+          if (index >= 0) {
```

Comment:
- Can't we inline with `headers.find`?
- Lowercase check is not happening for object mode

With currently increased logic, i would refactor header setting to a utility function